### ISSUE TITLE
"react-helmet" dependency is bad

### ISSUE THREAD

**Issue by birdofpreyru** [birdofpreyru/react-utils#156]
The way `react-helmet` works is not on par with the modern React. In particular it relies on `react-side-effect`, which I am not familiar to yet, but from the name and brief description I beileve it does in some workaround way the same stuff which should be achieved with context API and nested components. Or, more exactly, it allows to override tags from previous react-helmet issues by new instances anywhere in the code, but that's not a good way.

In other words: the amount of customizations done in this `react-utils` lib on top of `react-helmet` suggest it is better drop `react-helmet` alltogether and just re-write necessary pieces in a better way.

**Comment by birdofpreyru** [comment]
Presumably, the need for `react-helmet` will be eliminated in React v19

**Comment by birdofpreyru** [comment]
React v19 does not quite removes a need for meta-tags managing library. Perhaps, migrate to https://github.com/staylor/react-helmet-async &mdash; that seems to be a straightforward migration, and a reasonable alternative to the helmet, for now.

**Comment by birdofpreyru** [comment]
NOTE: react-helmet-async does not look like very actively maintained dependency either, but it should be fine for now.
